Subject: rt: sparseirq build fix
From: Ingo Molnar <mingo@elte.hu>
Date: Sun Feb 08 16:43:43 CET 2009

Convert it to raw locks:

kernel/irq/manage.c: In function ‘thread_simple_irq’:
kernel/irq/manage.c:821: error: ‘irq_desc’ undeclared (first use in this function)
kernel/irq/manage.c:821: error: (Each undeclared identifier is reported only once
kernel/irq/manage.c:821: error: for each function it appears in.)
kernel/irq/manage.c: In function ‘thread_level_irq’:
kernel/irq/manage.c:843: error: ‘irq_desc’ undeclared (first use in this function)
kernel/irq/manage.c: In function ‘thread_fasteoi_irq’:
kernel/irq/manage.c:855: error: ‘irq_desc’ undeclared (first use in this function)
kernel/irq/manage.c: In function ‘thread_edge_irq’:
kernel/irq/manage.c:867: error: ‘irq_desc’ undeclared (first use in this function)
kernel/irq/manage.c: In function ‘thread_do_irq’:
kernel/irq/manage.c:905: error: ‘irq_desc’ undeclared (first use in this function)
kernel/irq/manage.c: In function ‘init_hardirqs’:
kernel/irq/manage.c:1031: error: ‘irq_desc’ undeclared (first use in this function)
make[1]: *** [kernel/irq/manage.o] Error 1
make: *** [kernel/irq/manage.o] Error 2

=> fold back to: preempt-realtime-irqs.patch

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/irq/handle.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/kernel/irq/handle.c
===================================================================
--- linux-2.6-tip.orig/kernel/irq/handle.c
+++ linux-2.6-tip/kernel/irq/handle.c
@@ -79,7 +79,7 @@ static struct irq_desc irq_desc_init = {
 	.chip	    = &no_irq_chip,
 	.handle_irq = handle_bad_irq,
 	.depth      = 1,
-	.lock       = __SPIN_LOCK_UNLOCKED(irq_desc_init.lock),
+	.lock       = RAW_SPIN_LOCK_UNLOCKED(irq_desc_init.lock),
 };
 
 void init_kstat_irqs(struct irq_desc *desc, int cpu, int nr)
@@ -137,7 +137,7 @@ static struct irq_desc irq_desc_legacy[N
 		.chip	    = &no_irq_chip,
 		.handle_irq = handle_bad_irq,
 		.depth	    = 1,
-		.lock	    = __SPIN_LOCK_UNLOCKED(irq_desc_init.lock),
+		.lock	    = RAW_SPIN_LOCK_UNLOCKED(irq_desc_init.lock),
 	}
 };
 
